### PR TITLE
feat: improve hbase version

### PR DESCRIPTION
**Pull request by zyxxoo**

### CODE REVIEW

File: hugegraph-hbase/pom.xml [hunk 1]
```diff
@@ -34,14 +34,90 @@
             <version>${project.version}</version>
         </dependency>
         <dependency>
-            <groupId>com.baidu.hugegraph</groupId>
-            <artifactId>hbase-shaded-endpoint</artifactId>
-            <version>2.0.6</version>
+            <groupId>org.apache.hbase</groupId>
+            <artifactId>hbase-endpoint</artifactId>
+            <version>3.0.0-alpha-3</version>
```
Comment (imbajin) [inline hugegraph-hbase/pom.xml:39]: seems a little danger? the biggest version in `2.x` fail?

**Comment by javeme** [comment]
hbase ci error:
```c
2022-11-21 09:18:51 [main] [WARN] o.a.h.u.NativeCodeLoader - Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
2022-11-21 09:18:52 [Registry-endpoints-refresh-end-points] [INFO] o.a.h.h.c.RegistryEndpointsRefresher - Registry end points refresher loop exited.
Error: -21 09:18:52 [main] [ERROR] c.b.h.b.s.h.HbaseStore - Failed to open HBase 'm'
org.apache.hadoop.hbase.exceptions.MasterRegistryFetchException: Exception making rpc to masters [fv-az163-37.kzyshwy3mise5klwsllqwzgzyb.cx.internal.cloudapp.net,16000,-1]
	at java.lang.Thread.getStackTrace(Thread.java:1564) ~[?:1.8.0_352]
	at org.apache.hadoop.hbase.util.FutureUtils.setStackTrace(FutureUtils.java:130) ~[hbase-common-3.0.0-alpha-3.jar:3.0.0-alpha-3]
	at org.apache.hadoop.hbase.util.FutureUtils.rethrow(FutureUtils.java:149) ~[hbase-common-3.0.0-alpha-3.jar:3.0.0-alpha-3]
	at org.apache.hadoop.hbase.util.FutureUtils.get(FutureUtils.java:172) ~[hbase-common-3.0.0-alpha-3.jar:3.0.0-alpha-3]
	at org.apache.hadoop.hbase.client.ConnectionFactory.createConnection(ConnectionFactory.java:233) ~[hbase-client-3.0.0-alpha-3.jar:3.0.0-alpha-3]
	at org.apache.hadoop.hbase.client.ConnectionFactory.createConnection(ConnectionFactory.java:132) ~[hbase-client-3.0.0-alpha-3.jar:3.0.0-alpha-3]
	at com.baidu.hugegraph.backend.store.hbase.HbaseSessions.open(HbaseSessions.java:158) ~[classes/:?]
	at com.baidu.hugegraph.backend.store.hbase.HbaseStore.open(HbaseStore.java:163) ~[classes/:?]
	at com.baidu.hugegraph.StandardHugeGraph.clearBackend(StandardHugeGraph.java:354) ~[classes/:?]
	at com.baidu.hugegraph.core.CoreTestSuite.init(CoreTestSuite.java:69) ~[classes/:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_352]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_352]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_352]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_352]
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50) ~[junit-4.12.jar:4.12]
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12) ~[junit-4.12.jar:4.12]
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47) ~[junit-4.12.jar:4.12]
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24) ~[junit-4.12.jar:4.12]
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27) ~[junit-4.12.jar:4.12]
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363) ~[junit-4.12.jar:4.12]
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:365) ~[surefire-junit4-2.20.jar:2.20]
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:272) ~[surefire-junit4-2.20.jar:2.20]
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:236) ~[surefire-junit4-2.20.jar:2.20]
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:159) ~[surefire-junit4-2.20.jar:2.20]
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:386) ~[surefire-booter-2.20.jar:2.20]
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:323) ~[surefire-booter-2.20.jar:2.20]
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:143) ~[surefire-booter-2.20.jar:2.20]
	at --------Future.get--------(Unknown Source) ~[?:?]
	at org.apache.hadoop.hbase.client.AbstractRpcBasedConnectionRegistry.lambda$groupCall$1(AbstractRpcBasedConnectionRegistry.java:198) ~[hbase-client-3.0.0-alpha-3.jar:3.0.0-alpha-3]
	at org.apache.hadoop.hbase.util.FutureUtils.lambda$addListener$0(FutureUtils.java:68) ~[hbase-common-3.0.0-alpha-3.jar:3.0.0-alpha-3]
	at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:774) ~[?:1.8.0_352]
	at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:750) ~[?:1.8.0_352]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488) ~[?:1.8.0_352]
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1990) ~[?:1.8.0_352]
	at org.apache.hadoop.hbase.client.AbstractRpcBasedConnectionRegistry.lambda$call$0(AbstractRpcBasedConnectionRegistry.java:153) ~[hbase-client-3.0.0-alpha-3.jar:3.0.0-alpha-3]
	at org.apache.hbase.thirdparty.com.google.protobuf.RpcUtil$1.run(RpcUtil.java:79) ~[hbase-shaded-protobuf-4.1.0.jar:4.1.0]
	at org.apache.hbase.thirdparty.com.google.protobuf.RpcUtil$1.run(RpcUtil.java:70) ~[hbase-shaded-protobuf-4.1.0.jar:4.1.0]
	at org.apache.hadoop.hbase.ipc.AbstractRpcClient.onCallFinished(AbstractRpcClient.java:390) ~[hbase-client-3.0.0-alpha-3.jar:3.0.0-alpha-3]
	at org.apache.hadoop.hbase.ipc.AbstractRpcClient.access$100(AbstractRpcClient.java:92) ~[hbase-client-3.0.0-alpha-3.jar:3.0.0-alpha-3]
	at org.apache.hadoop.hbase.ipc.AbstractRpcClient$3.run(AbstractRpcClient.java:422) ~[hbase-client-3.0.0-alpha-3.jar:3.0.0-alpha-3]
	at org.apache.hadoop.hbase.ipc.AbstractRpcClient$3.run(AbstractRpcClient.java:417) ~[hbase-client-3.0.0-alpha-3.jar:3.0.0-alpha-3]
	at org.apache.hadoop.hbase.ipc.Call.callComplete(Call.java:114) ~[hbase-client-3.0.0-alpha-3.jar:3.0.0-alpha-3]
	at org.apache.hadoop.hbase.ipc.Call.setException(Call.java:129) ~[hbase-client-3.0.0-alpha-3.jar:3.0.0-alpha-3]
	at org.apache.hadoop.hbase.ipc.BufferCallBeforeInitHandler.userEventTriggered(BufferCallBeforeInitHandler.java:92) ~[hbase-client-3.0.0-alpha-3.jar:3.0.0-alpha-3]
	at org.apache.hbase.thirdparty.io.netty.channel.AbstractChannelHandlerContext.invokeUserEventTriggered(AbstractChannelHandlerContext.java:346) ~[hbase-shaded-netty-4.1.0.jar:?]
	at org.apache.hbase.thirdparty.io.netty.channel.AbstractChannelHandlerContext.invokeUserEventTriggered(AbstractChannelHandlerContext.java:332) ~[hbase-shaded-netty-4.1.0.jar:?]
	at org.apache.hbase.thirdparty.io.netty.channel.AbstractChannelHandlerContext.fireUserEventTriggered(AbstractChannelHandlerContext.java:324) ~[hbase-shaded-netty-4.1.0.jar:?]
	at org.apache.hbase.thirdparty.io.netty.channel.DefaultChannelPipeline$HeadContext.userEventTriggered(DefaultChannelPipeline.java:1428) ~[hbase-shaded-netty-4.1.0.jar:?]
	at org.apache.hbase.thirdparty.io.netty.channel.AbstractChannelHandlerContext.invokeUserEventTriggered(AbstractChannelHandlerContext.java:346) ~[hbase-shaded-netty-4.1.0.jar:?]
	at org.apache.hbase.thirdparty.io.netty.channel.AbstractChannelHandlerContext.invokeUserEventTriggered(AbstractChannelHandlerContext.java:332) ~[hbase-shaded-netty-4.1.0.jar:?]
	at org.apache.hbase.thirdparty.io.netty.channel.DefaultChannelPipeline.fireUserEventTriggered(DefaultChannelPipeline.java:913) ~[hbase-shaded-netty-4.1.0.jar:?]
	at org.apache.hadoop.hbase.ipc.NettyRpcConnection.failInit(NettyRpcConnection.java:198) ~[hbase-client-3.0.0-alpha-3.jar:3.0.0-alpha-3]
	at org.apache.hadoop.hbase.ipc.NettyRpcConnection.access$300(NettyRpcConnection.java:78) ~[hbase-client-3.0.0-alpha-3.jar:3.0.0-alpha-3]
	at org.apache.hadoop.hbase.ipc.NettyRpcConnection$2.operationComplete(NettyRpcConnection.java:287) ~[hbase-client-3.0.0-alpha-3.jar:3.0.0-alpha-3]
	at org.apache.hadoop.hbase.ipc.NettyRpcConnection$2.operationComplete(NettyRpcConnection.java:281) ~[hbase-client-3.0.0-alpha-3.jar:3.0.0-alpha-3]
	at org.apache.hbase.thirdparty.io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:578) ~[hbase-shaded-netty-4.1.0.jar:?]
	at org.apache.hbase.thirdparty.io.netty.util.concurrent.DefaultPromise.notifyListeners0(DefaultPromise.java:[571](https://github.com/apache/incubator-hugegraph/actions/runs/3513026795/jobs/5885375454#step:10:572)) ~[hbase-shaded-netty-4.1.0.jar:?]
	at org.apache.hbase.thirdparty.io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:550) ~[hbase-shaded-netty-4.1.0.jar:?]
	at org.apache.hbase.thirdparty.io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:491) ~[hbase-shaded-netty-4.1.0.jar:?]
	at org.apache.hbase.thirdparty.io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:616) ~[hbase-shaded-netty-4.1.0.jar:?]
	at org.apache.hbase.thirdparty.io.netty.util.concurrent.DefaultPromise.setFailure0(DefaultPromise.java:609) ~[hbase-shaded-netty-4.1.0.jar:?]
	at org.apache.hbase.thirdparty.io.netty.util.concurrent.DefaultPromise.tryFailure(DefaultPromise.java:117) ~[hbase-shaded-netty-4.1.0.jar:?]
	at org.apache.hbase.thirdparty.io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.fulfillConnectPromise(AbstractNioChannel.java:321) ~[hbase-shaded-netty-4.1.0.jar:?]
	at org.apache.hbase.thirdparty.io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:337) ~[hbase-shaded-netty-4.1.0.jar:?]
	at org.apache.hbase.thirdparty.io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:710) ~[hbase-shaded-netty-4.1.0.jar:?]
	at org.apache.hbase.thirdparty.io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:658) ~[hbase-shaded-netty-4.1.0.jar:?]
	at org.apache.hbase.thirdparty.io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:[584](https://github.com/apache/incubator-hugegraph/actions/runs/3513026795/jobs/5885375454#step:10:585)) ~[hbase-shaded-netty-4.1.0.jar:?]
	at org.apache.hbase.thirdparty.io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:496) ~[hbase-shaded-netty-4.1.0.jar:?]
	at org.apache.hbase.thirdparty.io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986) ~[hbase-shaded-netty-4.1.0.jar:?]
	at org.apache.hbase.thirdparty.io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[hbase-shaded-netty-4.1.0.jar:?]
	at org.apache.hbase.thirdparty.io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[hbase-shaded-netty-4.1.0.jar:?]
	at java.lang.Thread.run(Thread.java:750) [?:1.8.0_352]
Caused by: org.apache.hadoop.hbase.client.RetriesExhaustedException: Failed contacting masters after 1 attempts.
Exceptions:
java.net.ConnectException: Call to address=fv-az163-37.kzyshwy3mise5klwsllqwzgzyb.cx.internal.cloudapp.net:1[600](https://github.com/apache/incubator-hugegraph/actions/runs/3513026795/jobs/5885375454#step:10:601)0 failed on connection exception: org.apache.hbase.thirdparty.io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: fv-az163-37.kzyshwy3mise5klwsllqwzgzyb.cx.internal.cloudapp.net/10.1.0.196:16000

	at org.apache.hadoop.hbase.client.AbstractRpcBasedConnectionRegistry.lambda$groupCall$1(AbstractRpcBasedConnectionRegistry.java:197) ~[hbase-client-3.0.0-alpha-3.jar:3.0.0-alpha-3]
	... 43 more
2022-11-21 09:18:52 [Registry-endpoints-refresh-end-points] [INFO] o.a.h.h.c.RegistryEndpointsRefresher - Registry end points refresher loop exited.
```

**Comment by imbajin** [comment]
image

**Comment by imbajin** [comment]
seems big changes in `hbase3.x`  https://downloads.apache.org/hbase/3.0.0-alpha-3/api_compare_2.0.0_to_3.0.0-alpha-3RC0.html

image

File: hugegraph-hbase/pom.xml (hunk 1 above)
Comment (zyxxoo) [reply hugegraph-hbase/pom.xml:44]: yes, 2.x version use protobuf 2, conflict

**Pull request closed without merging**